app/services/track_commission_changes_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class TrackCommissionChangesService
+  include SolanaLogic
+
+  def initialize(current_batch: , network: "mainnet", solana_url: nil)
+    @current_batch = current_batch
+    @previous_batch = Batch.where("created_at < ?", @current_batch.created_at)
+                           .where(network: network)
+                           .where.not(scored_at: nil)
+                           .last
+    @network = network
+    @solana_url = solana_url || NETWORK_URLS[@network]
+  end
+
+  def call
+    VoteAccount.includes(:validator).where(network: @network).in_batches(of: 100) do |batch|
+      accounts = batch.pluck(:account)
+      result = get_inflation_rewards(accounts)
+
+      batch.each_with_index do |va, idx|  
+        next unless result[idx] # some accounts have no inflation reward e. g. if it was inactive
```

Comment:
You can use `compact_blank` on the active record relation instead.

Reply:
I don't want to do that, because then my result count would be different than my batch count and I wouldn't be able to assign results to vote accounts